GW2EIEvtcParser/EIData/ProfHelpers/Elementalist/ElementalistHelper.cs

```diff
@@ -400,7 +400,176 @@ internal static void ComputeProfessionCombatReplayActors(AbstractPlayer player,
+            {
+                var skill = new SkillModeDescriptor(player, Spec.Elementalist, FulgorSkill, SkillModeCategory.ShowOnSelect);
+                foreach (EffectEvent effect in fulgors)
+                {
+                    (long, long) lifespan = effect.ComputeLifespan(log, 4000);
+                    AddCircleSkillDecoration(replay, effect, color, skill, lifespan, 180, ParserIcons.EffectFulgor);
+                }
+            }
+
+            // Twister
+            if (log.CombatData.TryGetEffectEventsBySrcWithGUID(player.AgentItem, EffectGUIDs.ElementalistTwister, out IReadOnlyList<EffectEvent> twisters))
+            {
+                var skill = new SkillModeDescriptor(player, Spec.Elementalist, Twister, SkillModeCategory.CC);
+                foreach (EffectEvent effect in twisters)
+                {
+                    (long, long) lifespan = effect.ComputeDynamicLifespan(log, 1000);
+                    AddDoughnutSkillDecoration(replay, effect, color, skill, lifespan, 120, 240, ParserIcons.EffectTwister);
+                }
+            }
+
+            // Etching: Derecho
+            var etchingDerechoEffects = new[]
+            {
+                EffectGUIDs.ElementalistEtchingDerechoTier0,
+                EffectGUIDs.ElementalistEtchingDerechoTier1,
+                EffectGUIDs.ElementalistEtchingDerechoTier2,
+                EffectGUIDs.ElementalistEtchingDerechoTier3,
+                EffectGUIDs.ElementalistEtchingDerechoPerfect,
+            };
+            if (log.CombatData.TryGetEffectEventsBySrcWithGUIDs(player.AgentItem, etchingDerechoEffects, out IReadOnlyList<EffectEvent> etchingDerecho))
+            {
+                var skill = new SkillModeDescriptor(player, Spec.Elementalist, EtchingDerecho, SkillModeCategory.ShowOnSelect);
+                foreach (EffectEvent effect in etchingDerecho)
+                {
+                    (long, long) lifespan = effect.ComputeDynamicLifespan(log, 7000);
+                    AddCircleSkillDecoration(replay, effect, color, skill, lifespan, 360, ParserIcons.EffectEtchingDerecho);
+                }
+            }
+
+            // Fissure
+            if (log.CombatData.TryGetEffectEventsBySrcWithGUID(player.AgentItem, EffectGUIDs.ElementalistFissure, out IReadOnlyList<EffectEvent> fissures))
+            {
+                var skill = new SkillModeDescriptor(player, Spec.Elementalist, Fissure, SkillModeCategory.ShowOnSelect);
+                foreach (EffectEvent effect in fissures)
+                {
+                    (long, long) lifespan = effect.ComputeDynamicLifespan(log, 1000); // logged duration of 0 - overriding it to 1000 like the others
+                    AddDoughnutSkillDecoration(replay, effect, color, skill, lifespan, 120, 240, ParserIcons.EffectFissure);
+                }
+            }
+
+            // Etching: Haboob
+            var etchingHaboobEffects = new[]
```

Comment:
The code for the 4 etchings are basically identical, they can be regrouped under a private static AddEtchingDecorations API.

Reply:
Moved all, check if it's enough :D